extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/Cache.java

```diff
@@ -0,0 +1,14 @@
+package io.quarkus.cache.runtime;
+
+import java.util.concurrent.Callable;
+
+public interface Cache {
```

Comment:
Is that a contract a user is exposed to today? Or is it an implementation detail?
My fear is that this class (and similar) will leak into the project classpath and pollute it when someone does autocompletion. I imagine we will want to have a `Cache` contract later.

Reply:
This is an implementation detail (the contract a caching provider has to implement) that wouldn't be visible to the user if we were already using Java 9 modules.

Once again, it all depends on the possibility to introduce other caching providers in the future. If Caffeine is the only caching provider, then this interface is not required at all.

We could also rename it to `CacheImplementation` or something like that to avoid deciding about multiple caching providers for now.

Reply:
I removed the `Cache` interface for now. It will be easy to re-introduce later if we need it.